backend/pkg/api/instances.go

```diff
@@ -631,3 +639,110 @@ func (api *API) instanceStatusHistoryQuery(instanceID, appID, groupID string, li
 		Order(goqu.C("created_ts").Desc()).
 		Limit(uint(limit))
 }
+
+// instanceFactQuery returns a SelectDataset prepared to return all instances
+// that have been checked in during a given duration from a given time.
+func (api *API) instanceFactQuery(t *time.Time, duration *time.Duration) *goqu.SelectDataset {
+	if t == nil {
+		now := time.Now()
+		t = &now
+	}
+
+	if duration == nil {
+		d := time.Hour
+		duration = &d
+	}
+
+	// Convert duration to string
+	interval := fmt.Sprintf("%d milliseconds", int(duration.Milliseconds()))
+
+	return goqu.From("instance_application").
+		Select(
+			goqu.C("last_check_for_updates").As("timestamp"),
+			goqu.C("c.name").As("channel_name"),
+			goqu.Case().
+				When(goqu.C("c.arch").Eq(1), "AMD64").
+				When(goqu.C("c.arch").Eq(2), "ARM").
+				As("arch"),
+			goqu.C("version"),
+			goqu.L("COUNT(*)").As("instances")).
+		Join(goqu.T("groups"), goqu.On(goqu.C("group_id").Eq(goqu.C("groups.id")))).
+		Join(goqu.T("channel"), goqu.On(goqu.C("groups.channel_id").Eq(goqu.C("channel.id")))).
+		Join(goqu.T("package"), goqu.On(goqu.C("channel.package_id").Eq(goqu.C("package.id")))).
+		Where(
+			goqu.C("last_check_for_updates").Gt(goqu.L("?::timestamp - ?::interval", t.Format(time.RFC3339), interval)),
+			goqu.C("last_check_for_updates").Lte(t.Format(time.RFC3339))).
+		GroupBy("last_check_for_updates", "c.name", "c.arch", "version").
+		Order(goqu.C("last_check_for_updates").Desc())
+}
+
+// GetInstanceFacts returns an InstanceFact table with all instances that have
+// been previously been checked in.
+func (api *API) GetInstanceFacts() ([]InstanceFact, error) {
+	query := goqu.From("instance_fact").
+		Select(goqu.L("*"))
```

Comment:
```suggestion
		Select(goqu.L("*")).Order(goqu.C("timestamp").Asc())
```
Something like that to order

Reply:
I've sorted in descending order for the previous function, would the most recent timestamps not be more relevant to include at the top of the list?

Reply:
For going through the historic data I think it's better to start from the left of the x axis but yeah, it's just my preference and for me it would be more readable in a debug output to see oldest first and then how the statistics evolve related to when our test instances check in.